diff/parse.go

```diff
@@ -223,8 +230,31 @@ func (r *FileDiffReader) HunksReader() *HunksReader {
 
 // ReadFileHeaders reads the unified file diff header (the lines that
 // start with "---" and "+++" with the orig/new file names and
-// timestamps).
+// timestamps). Or starts with "Only in " message with dir path filename.
 func (r *FileDiffReader) ReadFileHeaders() (origName, newName string, origTimestamp, newTimestamp *time.Time, err error) {
+	if (r.fileHeaderLine != nil) && (bytes.HasPrefix(r.fileHeaderLine, onlyInMessagePrefix)) {
+		path := bytes.Split(bytes.TrimPrefix(
+			bytes.TrimSuffix(r.fileHeaderLine, []byte("\n")),
+			onlyInMessagePrefix), []byte(":"))
+		if len(path) != 2 {
+			return "", "", nil, nil,
+				&ParseError{r.line, r.offset, ErrBadOnlyInMessage}
+		}
+
+		source, filename := string(bytes.TrimSpace(path[0])), string(bytes.TrimSpace(path[1]))
+
+		unquotedSource, err := strconv.Unquote(source)
+		if err == nil {
+			source = unquotedSource
+		}
+		unquotedFilename, err := strconv.Unquote(filename)
+		if err == nil {
+			filename = unquotedFilename
+		}
+
+		return filepath.Join(source, filename), "", nil, nil, nil
```

Comment:
Do we not have to interpret `source` to figure out if `filename` was added or removed and depending on that `return "", filepath.Join(source, filename), nil, nil, nil` instead?

Reply:
All filenames from `Only in {source}: {filename}` messages will be added to `OrigName` field.